docs/DevGuide/Variables.md

```diff
@@ -0,0 +1,75 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Using Variables in SpECTRE
+
+# What is a Variables and Why Use Them?
+Variables are a data structure that allow you to make most if not all memory
+allocations needed at one time. Allocating memory can be expensive as it holds
+up any computations while the memory is being allocated. Using a Variables to
```

Comment:
You could also mention that, because the physical memory is shared between CPU cores, memory allocation is inherently serial and so for large numbers of cores this becomes a serial bottleneck in applications.